src/subcommand/init.rs

```diff
@@ -160,15 +160,11 @@ function {}
         or return $status
 
     else
-        # FIXME: use string-collect from fish 3.1.0 once it has wider adoption
-        set -l IFS ''
```

Comment:
By default, fish scans newlines as spaces, and it does not print `z --help` properly. This would have been fixed by `string-collect`, but temporarily setting `IFS=''` also fixes the problem.